Sources/SourceKitLSP/Swift/Diagnostic.swift

```diff
@@ -71,10 +59,46 @@ extension CodeAction {
   }
 
   init?(_ fixIt: FixIt, in snapshot: DocumentSnapshot) {
-    // FIXME: Once https://github.com/apple/swift-syntax/pull/2226 is merged and
-    // FixItApplier is public, use it to compute the edits that should be
-    // applied to the source.
-    return nil
+    var textEdits = [TextEdit]()
+    for edit in fixIt.edits {
+      guard let startPosition = snapshot.position(of:edit.range.lowerBound),
+        let endPosition = snapshot.position(of:edit.range.upperBound) else {
+        continue
+      }
+      textEdits.append(TextEdit(range: startPosition..<endPosition, newText: edit.replacement))
+    }
+
+    guard let generatedTitle = Self.generateTitle(with: textEdits, snapshot: snapshot) else {
+      return nil
+    }
+
+    self.init(
+      title: generatedTitle,
+      kind: .quickFix,
+      diagnostics: nil,
+      edit: WorkspaceEdit(changes: [snapshot.uri: textEdits])
+    )
+  }
+
+  private static func generateTitle(with edits: [TextEdit], snapshot: DocumentSnapshot) -> String? {
```

Comment:
Nitpick, I would name this without `generate`. Since this is side-effect free, I don’t think the function name needs a result and `generate` basically just says that the function is returning something, which is obvious from the result type.
```suggestion
  private static func title(for edits: [TextEdit], in snapshot: DocumentSnapshot) -> String? {
```

Reply:
you're right. let me update it.